components/plugin_marketplace/marketplace_item/marketplace_item.tsx

```diff
@@ -95,304 +60,23 @@ export const Label = ({name, description, url, color}: MarketplaceLabel): JSX.El
-            </p>,
-        );
-    }
-
-    let sameMajorVersion = false;
-    try {
-        sameMajorVersion = semver.major(version) === semver.major(installedVersion);
-    } catch (e) {
-        // If we fail to parse the version, assume a potentially breaking change.
-        // In practice, this won't happen since we already tried to parse the version above.
-    }
-
-    if (!sameMajorVersion) {
-        if (releaseNotesUrl) {
-            messages.push(
-                <p
-                    className='alert alert-warning'
-                    key='warning'
-                >
-                    <FormattedMarkdownMessage
-                        id='marketplace_modal.list.update_confirmation.message.warning_major_version_with_release_notes'
-                        defaultMessage='This update may contain breaking changes. Consult the [release notes](!{releaseNotesUrl}) before upgrading.'
-                        values={{releaseNotesUrl}}
-                    />
-                </p>,
-            );
-        } else {
-            messages.push(
-                <p
-                    className='alert alert-warning'
-                    key='warning'
-                >
-                    <FormattedMessage
-                        id='marketplace_modal.list.update_confirmation.message.warning_major_version'
-                        defaultMessage={'This update may contain breaking changes.'}
-                    />
-                </p>,
-            );
-        }
-    }
-
-    return (
-        <ConfirmModal
-            show={show}
-            title={
-                <FormattedMessage
-                    id='marketplace_modal.list.update_confirmation.title'
-                    defaultMessage={'Confirm Plugin Update'}
-                />
-            }
-            message={messages}
-            confirmButtonText={
-                <FormattedMessage
-                    id='marketplace_modal.list.update_confirmation.confirm_button'
-                    defaultMessage='Update'
-                />
-            }
-            onConfirm={onUpdate}
-            onCancel={onCancel}
-        />
-    );
-};
-
 export type MarketplaceItemProps = {
     id: string;
     name: string;
     description?: string;
-    version: string;
-    homepageUrl?: string;
-    releaseNotesUrl?: string;
-    labels?: MarketplaceLabel[];
     iconData?: string;
-    installedVersion?: string;
-    installing: boolean;
-    error?: string;
-    isDefaultMarketplace: boolean;
-    trackEvent: (category: string, event: string, props?: any) => void;
+    labels?: MarketplaceLabel[];
+    homepageUrl?: string;
 
-    actions: {
-        installPlugin: (category: string, event: string) => void;
-        closeMarketplaceModal: () => void;
-    };
-};
+    error?: string;
 
-type MarketplaceItemState = {
-    showUpdateConfirmationModal: boolean;
+    button: JSX.Element;
+    updateDetails: JSX.Element | null;
+    versionLabel: JSX.Element| null;
```

Comment:
That seems the most elegant way to me. Open for other suggestion on how to do the composition here.